openwisp_radius/api/serializers.py

```diff
@@ -126,10 +132,39 @@ def get_pdf_link(self, obj):
     def validate(self, data):
         if data['strategy'] == 'prefix' and not data.get('number_of_users'):
             raise serializers.ValidationError(
-                {'number_of_users': 'The field number_of_users cannot be empty'}
+                {'number_of_users': 'The field number_of_users cannot be empty.'}
+            )
+        try:
+            self.batch_org = Organization.objects.get(slug=data['organization_slug'])
+        except Organization.DoesNotExist:
+            raise serializers.ValidationError(
+                {'organization_slug': 'Organization not found.'}
+            )
+        try:
+            RadiusBatch.objects.get(name=data['name'], organization=self.batch_org)
+        except RadiusBatch.DoesNotExist:
+            pass
+        else:
+            raise serializers.ValidationError(
+                {'name': 'RadiusBatch with same name exists for organization.'}
```

Comment:
Look at this: https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/base/models.py#L844-L865

Can we reuse that validation logic in this serializer? 

Maybe you don't know it, but DRF at some point changed the validation and stopped calling full_clean(), the argument is that API validation can be different from model validation.
I never loved this, and that's why we have [ValidatedModelSerializer in openwisp-utils](https://github.com/openwisp/openwisp-utils#openwisp-utils-api-serializers-validatedmodelserializer). Can you try adding it to this serializer? It could be a good chance to remove duplication and add more consistency to the codebase (ideally we should have all serializers based on `ValidatedModelSerializer` but we can look into that later on).